src/vcpkg/commands.build.cpp

```diff
@@ -752,18 +774,23 @@ namespace vcpkg
         std::vector<CMakeVariable> variables{
             {"ALL_FEATURES", all_features},
             {"CURRENT_PORT_DIR", scfl.port_directory()},
-            {"_HOST_TRIPLET", action.host_triplet.canonical_name()},
+            {"_HOST_TRIPLET", host_triplet.canonical_name()},
             {"FEATURES", Strings::join(";", action.feature_list)},
             {"PORT", port_name},
             {"VERSION", scf.to_version().text},
-            {"VCPKG_USE_HEAD_VERSION", Util::Enum::to_bool(action.build_options.use_head_version) ? "1" : "0"},
-            {"_VCPKG_DOWNLOAD_TOOL", to_string_view(action.build_options.download_tool)},
-            {"_VCPKG_EDITABLE", Util::Enum::to_bool(action.build_options.editable) ? "1" : "0"},
-            {"_VCPKG_NO_DOWNLOADS", !Util::Enum::to_bool(action.build_options.allow_downloads) ? "1" : "0"},
+            {"VCPKG_USE_HEAD_VERSION",
+             action.request_type == RequestType::USER_REQUESTED && Util::Enum::to_bool(build_options.use_head_version)
+                 ? "1"
+                 : "0"},
+            {"_VCPKG_DOWNLOAD_TOOL", to_string_view(build_options.download_tool)},
+            {"_VCPKG_EDITABLE",
+             action.request_type == RequestType::USER_REQUESTED && Util::Enum::to_bool(build_options.editable) ? "1"
```

Comment:
I can see an argument to put UseHeadVersion and/or Editable back into the ActionPlan since this can vary by according to user requested but this seemed simpler.

Reply:
That's a good reason to go that way instead.